pkg/identity/cache/allocator.go

```diff
@@ -264,6 +264,10 @@ func (m *CachingIdentityAllocator) Close() {
 	}
 
 	m.IdentityAllocator.Delete()
+	if m.events != nil {
+		close(m.events)
+	}
```

Comment:
I wonder if these lines should be deleted altogether. 

The events channel gets initialized in the [`NewCachingIdentityAllocator` function](https://github.com/cilium/cilium/blob/master/pkg/identity/cache/allocator.go#L230-L241). According to the comments of the `Close` function, once it is executed, it should be possible to call `InitIdentityAllocator` again, which though uses the events channel. Hence, it would break, given that the channel would be closed. It doesn't seem to me that `Close` is ever used outside of tests, so no actual difference should be noticed, still, I feel it might be better not to close that channel to better comply with the desired approach.

Reply:
I don't see any particularly clean ways for handling this (at the moment it does not lead to issues, since the allocator is never initialized again after having been closed). I wonder if the other code reviewers have any suggestion.

Reply:
yeah you are right about the comment, hmm.. if anybody sees a potential problem with this I can just revert this. The oneliner that fixes the memleak here is sufficient as well (`allocator.WithoutGC()`)

Reply:
I think the code comment is moot; I see no evidence of Close() being called except for on daemon shutdown. I think we can just remove that comment and, if desired, panic on `InitIdentityAllocator` being called twice.

The real question: would this change be safe to backport? I'm checking.